### ISSUE TITLE
World Creation Crash w/ Custom Added Modded Biomes

### ISSUE THREAD

**Issue by WenXin20** [BlackJar72/ClimaticBiomePlacement#14]
Forge v14.23.5.2807-1.12.2
Climatic Biomes v2.6.0
[Mubble](https://minecraft.curseforge.com/projects/mubble) v1.3-pre4

I get this crash whenever I add the following biomes to the config. The biome names are the same as the ones listed in the BiomeList.txt. At first I thought it had to do with the space in the biome id, but then the landfill biome from [Dumpster Diving](https://minecraft.curseforge.com/projects/dumpster-diving) also crashed it.  Or am I doing something incorrectly?
`biome(mubble:mushroom kingdom)`
`seed(mubble:mushroom kingdom, 20, biomesoplenty:tropical_island)`  
in the SpecialIslandWarm.cfg in the custom folder

The same crash also happens with this:
`biome(dumpsterdiving:landfill)`
in the plains.cfg in the custom folder

[Crash Report](https://paste.dimdev.org/ixocuxiyoc.mccrash)

**Comment by BlackJar72** [comment]
I see the problem -- "mubble:mushroom kingdom" doesn't work because it treats spaces as delimiters, so it's trying to create an entry where "mubble:mushroom" is followed by the number "kingdom" and then the name "20."  I could probably re-work it to using the parser I made for Doomlike Dungeons and allowing strings with space to be put in quotes.  I'm not at home now, but maybe I could do this in the evening.

Personally, I think whoever made that biome made the mistake by putting white space inside the resource name, but I'll try to support it.

**Comment by BlackJar72** [comment]
Try the new 2.6.1 version and let me know if that fixes your problem.  You will need to put quotes around all biome names with spaces in them for it to work.

EDIT: If you still have trouble with the dumpster diving mod you might want to bring it up there; it shouldn't be affected by this.

**Comment by WenXin20** [comment]
I still get the same crash.
These crashes:
`"mubble:mushroom kingdom"`
`dumpsterdiving:landfill`
But this one from the same mod works, even without quotes.
`"mubble:scarlet forest"`

**Comment by BlackJar72** [comment]
Well, that doesn't make a lot of sense.  Could you post the actual error reports for each and the config files you're using? -- maybe even the biome lists.  If I can make it work it at all (no promises at this point), that could be very helpful.

**Comment by BlackJar72** [comment]
Capitalize "Mushroom Kingdom" ("mubble:Mushroom Kingdom") for one thing (EDIT: That's how it is in the [Mubble source code](https://github.com/Hugman76/Mubble/blob/master/java/hugman/mod/init/BiomeInit.java)). I don't see any scarlet forest in the Mubble source code.  I've contacted the Mubble author for clarification on this.

 I'm not sure about landfill; that mod does not have available source code.

You are putting each entry on its own line, right?

**Comment by Hugman76** [comment]
Hello there!

My mod actually uses the Scarlet Forest, and since the version is Mubble v1.3-pre4 it means it is actually a beta and by further more the source code for my betas is located on the dev branch. So the file is [here](https://github.com/Hugman76/Mubble/blob/dev/java/hugman/mod/init/MubbleBiomes.java). Also, it seems like I did pretty much a big mistake by actually doing the registry name of the biomes, because at the time I thinked this particular string was actually the name that would show up and not the internal ID.

Further more, I fixed this issue for v1.3-pre5, that will air soon. The biome IDs will actually be `mubble:mushroom_kingdom` and `mubble:scarlet_forest`.

Thank you!
